quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java

```diff
@@ -106,6 +103,18 @@ private DatabasePropertyMappers(){}
         };
     }
 
+    private static String getXaOrNonXaDriver(String db, ConfigSourceInterceptorContext context) {
+        ConfigValue xaEnabledConfigValue = context.proceed("kc.transaction-xa-enabled");
+
+        boolean isXaEnabled = xaEnabledConfigValue == null || Boolean.parseBoolean(xaEnabledConfigValue.getValue());
+
+        if (isXaEnabled) {
```

Comment:
```suggestion
        return Database.getDriver(db, isXaEnabled).orElse(db);
```

Should be simpler. But not a blocker.